feat: use new bip44 token selectors

bergarces opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

Use new BIP44 selectors to create an ordered list that contains all assets, as well as new list item component, based on the existing one, that uses the data from the new selector. 

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

bergarces committed:
progress

bergarces committed:
merge main

bergarces committed:
updates

bergarces committed:
fix asset to token mapping for backwards compatibility

bergarces committed:
progress

bergarces committed:
update package

bergarces committed:
remove $ from fiat value

bergarces committed:
fixed sorting

bergarces committed:
display  solana

bergarces committed:
correction for networks

bergarces committed:
registry

bergarces committed:
progress

bergarces committed:
update lockfile

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
simplify sorting

bergarces committed:
add symbol to currency

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
merge main

bergarces committed:
fix network filtering

bergarces committed:
move selectors

bergarces committed:
remove hardcoded bip44 flag

bergarces committed:
code fence and updated package

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
tests for selector

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
update package

bergarces committed:
add staked assets

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
staked assets

bergarces committed:
todo comments

bergarces committed:
remove hardcoded flag

bergarces committed:
remove console.log

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
better use of selectors

bergarces committed:
tests for selectors

bergarces committed:
all tests for selectors

bergarces committed:
update token list item

bergarces committed:
tests for list item component

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces committed:
remove changes

bergarces committed:
improved tests

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces commented on app/components/UI/Tokens/TokenList/TokenListItem/TokenListItemBip44.test.tsx:
The tests on `TokenListItem` were a bit odd, most of them extracting random component logic, testing it in isolation and then rendering the component for coverage with `expect(true).toBe(true)`.

Ideally, we would want to extract avatar, network badge and percentage logic outside of the component and test it in isolation, as that is being used in many places, but I believe that's out of scope here.

That would leave us with a very small and easy to test list item component.

bergarces committed:
comments

bergarces commented on app/components/UI/Tokens/TokenList/TokenListItem/TokenListItemBip44.tsx:
Mostly copied `TokenListItem` and stripped all the formatting and balance calculation logic that is no longer needed.

Prithpal-Sooriya replied:
Okay sounds good.

I think we can do some cleanup and perf next week and during RC.

Prithpal-Sooriya commented on app/components/UI/Tokens/TokenList/index.tsx:
```diff
@@ -48,6 +49,14 @@ const TokenListComponent = ({
+  const isMultichainAccountsState2Enabled = useSelector(
+    selectMultichainAccountsState2Enabled,
+  );
+  const TokenListItemComponent = isMultichainAccountsState2Enabled
+    ? TokenListItemBip44
+    : TokenListItem;
```
Great cut point, this way we have a dedicated component with less feature flags to wrangle.

bergarces commented on app/components/UI/Tokens/TokenList/TokenListItem/TokenListItemBip44.tsx:
```diff
@@ -0,0 +1,298 @@
+      Number.isFinite(pricePercentChange1d);
+
+    // Determine the color for percentage change
+    let percentageColor = TextColor.Alternative;
+    if (hasPercentageChange) {
+      if (pricePercentChange1d === 0) {
+        percentageColor = TextColor.Alternative;
+      } else if (pricePercentChange1d > 0) {
+        percentageColor = TextColor.Success;
+      } else {
+        percentageColor = TextColor.Error;
+      }
+    }
+
+    const percentageText = hasPercentageChange
+      ? `${pricePercentChange1d >= 0 ? '+' : ''}${pricePercentChange1d.toFixed(
+          2,
+        )}%`
+      : undefined;
```
These changes were recently introduced in this PR. I just copied them over.
https://github.com/MetaMask/metamask-mobile/pull/18289/files#diff-6d5005ff2f66538d8c8facca733a5cee4827ffcdac3a4c2251c5fd1c047892e4

bergarces committed:
renderWithProvider

bergarces committed:
remove more code from new component

bergarces committed:
Merge branch 'main' into bip44-selector-for-tokens

bergarces commented:
I have removed some extra duplicated code from the new component.

One of the functions to create the network logo was already exported from `Balances`. For the asset logo, I have extracted the logic to a new component with its own style an test.

bergarces committed:
correction

bergarces commented:
Due to current CI issues with iOS tests, and after passing all Android tests, I'm skipping smoke E2E so it can be merged.
